src/local/Matcher.ts

```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash';
+
+interface Data<T = undefined> {
```

Comment:
Is `data` really allowed to be `undefined`? Do we have any expectations of what T must be that T should extend?

Reply:
I copied that type from the old `d.ts` file, don't know what the original reasoning was.